sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/restriction-bypasses-resolution-step.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from "@nestjs/common";
+import { NoteSharedService, SystemUsersService } from "@sims/services";
+import {
+  ApplicationRestrictionBypass,
+  NoteType,
+  RestrictionBypassBehaviors,
+} from "@sims/sims-db";
+import { ECertProcessStep } from "./e-cert-steps-models";
+import { EntityManager } from "typeorm";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { EligibleECertDisbursement } from "../disbursement-schedule.models";
+
+/**
+ * Check if restrictions bypasses are active and configured as {@link RestrictionBypassBehaviors.NextDisbursementOnly}.
+ * All active bypasses identified should be removed.
+ */
+@Injectable()
+export class RestrictionBypassesResolutionStep implements ECertProcessStep {
+  constructor(
+    private readonly noteSharedService: NoteSharedService,
+    private readonly systemUsersService: SystemUsersService,
+  ) {}
+
+  /**
+   * Remove restrictions active bypasses configured as {@link RestrictionBypassBehaviors.NextDisbursementOnly}.
+   * @param eCertDisbursement eligible disbursement to be potentially added to an e-Cert.
+   * @param entityManager used to execute the commands in the same transaction.
+   * @param log cumulative log summary.
+   */
+  async executeStep(
+    eCertDisbursement: EligibleECertDisbursement,
+    entityManager: EntityManager,
+    log: ProcessSummary,
+  ): Promise<boolean> {
+    log.info("Checking if restrictions bypasses should be removed.");
+    const nextDisbursementOnlyBypasses =
+      eCertDisbursement.activeRestrictionBypasses.filter(
+        (bypass) =>
+          bypass.bypassBehavior ===
+          RestrictionBypassBehaviors.NextDisbursementOnly,
+      );
```

Comment:
Let me know if I am missing something here, are we checking if the bypass is relevant to the e-cert. Or is it even required to check that.

for e.g. Consider there is a e-cert for part time which on the completion of all it's steps can set a bypass applied for a restriction which has stop full-time disbursement to inactive.

Reply:
I am not sure if I follow the question. Trying to explain the process next.
1 - `activeRestrictionBypasses` contains all active bypasses associated with the application and they belong to that application context only.
2 - At the end of all steps, any active bypass with the behavior `NextDisbursementOnly` should be removed.
So, we select all active bypasses to be removed and update them.
Please keep in mind that bypasses are applied per application and offering intensity.
Hope that helps, otherwise, we can have a call. Please let me know.

Reply:
Oh yes, I get it now. It is application based, so all good. Somehow I lost the way considered it to be student based.